python/python/lance/vector.py

```diff
@@ -115,3 +118,40 @@ def vec_to_table(
             f"data must be dict, list, or ndarray, got {type(data)} instead"
         )
     return pa.Table.from_arrays(arrays, names=names)
+
+
+def train_ivf_centroids(
+    dataset: LanceDataset,
+    column: str,
+    k: int,
+    metric_type: str,
+    accelerator: str,
+    *,
+    sample_rate: int = 256,
+) -> np.ndarray:
+    """Use accelerator (GPU or MPS) to train kmeans."""
+
+    if accelerator not in ["gpu", "cuda"]:
+        raise ValueError(
+            "Train ivf centroids on accelerator: "
+            + f"only support 'cuda' as accelerator, got '{accelerator}'."
+        )
+
+    total_size = dataset.count_rows()
+    sample_size = min(k * sample_rate, total_size)
+
+    if total_size < k * sample_size:
+        samples = dataset.to_table(columns=[column])[column].combine_chunks()
+    else:
+        samples = dataset.sample(k * sample_rate)[column]
+
+    if accelerator in ["gpu", "cuda"]:
```

Comment:
nit: gpu could be MPS or AMD as well, I don't think we should map `gpu` to `cuda`.

Reply:
Yea, so i was thinking that use `gpu` to call `preferred_device()` later to auto detect GPU on the machine

https://github.com/lancedb/lance/blob/fe4b17943813dfab0d799be6990d9fc93f3f0cea/python/python/lance/torch/__init__.py#L23

But it is fair that we don't need to do it now. As mps performance is not good at the moment.